forest/sycamore/survey_config.py

```diff
@@ -22,75 +25,114 @@ def convert_time_to_date(submit_time, day, time):
+        raise ValueError('Incorrect type of survey.'
+                         ' Ensure this is weekly, absolute, or relative.')
+
+    if timings is None:
+        timings = []
 
         # Get the number of weeks between start and end time
     t_start = pd.Timestamp(time_start)
     t_end = pd.Timestamp(time_end)
-
-    weeks = pd.Timedelta(t_end - t_start).days
-    # Get ceiling number of weeks
-    weeks = math.ceil(weeks / 7.0)
-
-    # Roll dates
-    t_lag = list(np.roll(np.array(timings, dtype="object"), -1))
-
-    # for each week, generate the survey times and append to a list
-    start_dates = [time_start + datetime.timedelta(days=7 * (i)) for i in range(weeks)]
-
     surveys = []
 
-    for s in start_dates:
-        # Get the starting day of week 
-        #         dow_s = s.weekday()
-        for i, t in enumerate(t_lag):
-            if len(t) > 0:
-                surveys.extend(convert_time_to_date(s, day=i, time=t))
-
+    if survey_type == 'weekly':
+
+        weeks = pd.Timedelta(t_end - t_start).days
+        # Get ceiling number of weeks
+        weeks = math.ceil(weeks / 7.0)
+
+        # Roll dates
+        t_lag = list(np.roll(np.array(timings, dtype="object"), -1))
+
+        # for each week, generate the survey times and append to a list
+        start_dates = [
+            t_start + datetime.timedelta(days=7 * i) for i in range(weeks)
+        ]
+
+        for s in start_dates:
+            # Get the starting day of week
+            #         dow_s = s.weekday()
+            for i, t in enumerate(t_lag):
+                if len(t) > 0:
+                    surveys.extend(convert_time_to_date(s, day=i, time=t))
+    if survey_type == 'absolute':
+        times_df = pd.DataFrame(timings)
+        times_df.columns = ['year', 'month', 'day', 'second']
+        surveys = pd.to_datetime(times_df).tolist()
+
+    if survey_type == 'relative':
+        if intervention_dict is None:
+            logger.error('Error: No dictionary of interventions provided.')
+            return []
+        for t in timings:
+            # This could probably be vectorized
+            if intervention_dict[t[0]] is not None:  # time exists for the user
+                # first, get the intervention time.
+                # Then, add the days and time of day for the survey to the time
+                current_time = (
+                    datetime.datetime.fromisoformat(intervention_dict[t[0]])
+                    + pd.Timedelta(t[1], unit='days')
+                    + pd.Timedelta(t[2], unit='seconds')
+                )  # seconds after time
+                surveys.append(current_time)
     return surveys
 
 
-def gen_survey_schedule(config_path, time_start, time_end, beiwe_ids):
-    """
+def gen_survey_schedule(
+        config_path: str, time_start: str, time_end: str, beiwe_ids: list,
+        all_interventions_dict: dict
+                        ) -> pd.DataFrame:
```

Comment:
Sorry, I should have been more clear. It should be just:
```
def gen_survey_schedule(
    config_path: str, time_start: str, time_end: str, beiwe_ids: list,
    all_interventions_dict: dict
) -> pd.DataFrame:
```
or
```
def gen_survey_schedule(config_path: str,
                        time_start: str,
                        time_end: str,
                        beiwe_ids: list,
                        all_interventions_dict: dict) -> pd.DataFrame:
```
or
```
def gen_survey_schedule(
        config_path: str,
        time_start: str,
        time_end: str,
        beiwe_ids: list,
        all_interventions_dict: dict
) -> pd.DataFrame:
```

Reply:
Thanks for the clarification!